File: go/src/leetcode/lc0007_test.go
package leetcode

import "testing"

func Test_reverse(t *testing.T) {
	test0007(123, 321, t)
	test0007(-123, -321, t)
	test0007(120, 21, t)
}

func test0007(x int, expected int, t *testing.T) {
	if expected != reverse(x) {
		t.Error()
	}
}

File: py/leetcode/lc0007_test.py
import unittest
import leetcode.lc0007 as lc


class Test(unittest.TestCase):
    def setUp(self):
        self.solution = lc.Solution()

    def test(self):
        self._test(123, 321)
        self._test(-123, -321)
        self._test(120, 21)

    def _test(self, x, expected):
        self.assertEqual(expected, self.solution.reverse(x))
